### ISSUE TITLE
Implement SRL in Java

### ISSUE THREAD

**Issue by amalic** [TYVRNET/SRL#11]
As a Java developer, I would love to see a Java implementation of SRL.

**Comment by SamTebbs33** [comment]
i wouldn't midn contributing to something like this, especially if it were written in Scala.

**Comment by KarimGeiger** [comment]
That's great news :) Sadly, I'm not quite familiar with Scala or Java. I hope someone makes the first step, though.

**Comment by KarimGeiger** [comment]
In the future, I'll be moving this project to [SimpleRegex/PHP](https://github.com/SimpleRegex/PHP), to have a better overview of everything to come. 

If someone wants to contribute an official Java/Scala version of this, please let me know. I'll then create a repository at [SimpleRegex/Java](https://github.com/SimpleRegex/Java) or [SimpleRegex/Scala](https://github.com/SimpleRegex/Scala) and give you permission on it.

**Comment by SamTebbs33** [comment]
It would be best to first of all make a Java version and then create a Scala wrapper around that. If you make the repo then please add me and I will have a look through your PHP code to get a picture of how the system works.

**Comment by SamTebbs33** [comment]
Do you have a preferred git workflow to use in the Java repo? I tend to use a version of git-flow.

**Comment by KarimGeiger** [comment]
Yes, git-flow would be a great choice.

**Comment by KarimGeiger** [comment]
FYI: I've renamed the project. to SRL-Java.

**Comment by SamTebbs33** [comment]
Would this approach be better than copying over the files from the current repo to the SimpleRegex/SRL-PHP repo? https://help.github.com/articles/about-repository-transfers/

**Comment by SamTebbs33** [comment]
When you migrate a repo you don't lose the issues et.c, but I see your point for not breaking links to the current repo.

**Comment by marcosvega91** [comment]
Hi , If you want I can help you with the implementation of the library in java :). Let me know

**Comment by KarimGeiger** [comment]
Sure, that's great, @marcosvega91! I've added you to the group.

**Comment by marcosvega91** [comment]
Yeah thank you :)

**Comment by SamTebbs33** [comment]
@KarimGeiger Could you make a gitter channel for the repo/s? It'll be easier to talk that way :)

**Comment by marcosvega91** [comment]
@SamTebbs33 In lack of that I have contacted you on facebook :)

**Comment by KarimGeiger** [comment]
I was just trying to invite you and @marcosvega91 to the slack team at https://simpleregex.slack.com

Would slack be okay too? :)

**Comment by SamTebbs33** [comment]
Yeah sure, I just get annoyed by their registration system :stuck_out_tongue_closed_eyes:

**Comment by KarimGeiger** [comment]
That's true.. I followed you on Twitter. You can drop me your email as a direct message, if you like.

@marcosvega91, how can I get you on board?

**Comment by hoshsadiq** [comment]
I wouldn't mind helping out as well

**Comment by KarimGeiger** [comment]
Great! I've added you to the Java team. They started working on the dev branch already. And if you drop me an email at geiger@karim.email, I'll add you to the Slack team as well.

**Comment by hoshsadiq** [comment]
Cheers. Sent you a mail!

**Comment by amalic** [comment]
any progress on the java/scala implementation?
